src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx

```diff
@@ -157,6 +161,32 @@ const MainInfo = ({ data }: MainInfoProps) => {
     />
   );
 
+  const switcher = <Switcher checked={showUSD} onChange={() => setShowUSD(!showUSD)} />;
+
+  function aptOrUsd(value: bigint, valueUsd: number | undefined) {
+    if (showUSD) {
+      if (valueUsd) {
+        return (
+          <>
+            <FormattedNumber value={valueUsd} scramble />
+            &nbsp;
+            {"$"}
+          </>
+        );
+      } else {
+        return <>{"N/A $"}</>;
+      }
+    } else {
+      return (
+        <>
+          <FormattedNumber value={value} nominalize scramble />
+          &nbsp;
+          <AptosIconBlack className={"icon-inline mb-[0.3ch]"} />
+        </>
+      );
+    }
+  }
+
```

Comment:
```suggestion
  const switcher = <Switcher disabled={valueUsd === undefined} checked={showUSD} onChange={() => setShowUSD((v) => !v)} />;

  function aptOrUsd(value: bigint, valueUsd: number | undefined) {
    if (valueUsd !== undefined && showUSD) {
      return (
        <>
          <FormattedNumber value={valueUsd} scramble />
          {" $"}
        </>
      );
    } else {
      return (
        <>
          <FormattedNumber value={value} nominalize scramble />
          {" "}
          <AptosIconBlack className={"icon-inline mb-[0.3ch]"} />
        </>
      );
    }
  }

```

Make sure to change to `setShowUSD((v) => !v)` bc it avoids issues with closures and stale state. It probably doesn't matter here, but if there are multiple state updates for a single render, the final render might not use the last value unless you do it like this

The other stuff is just to disable the toggle and not worry about displaying it if there's no USD value